src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,286 @@
+                alt={media.alt}
+                size="xxxsmall"
+                sx={{ borderRadius: "24px" }}
+              />
+            );
+          }
+        }
+        return null;
+      };
+
+      /** Renders the trigger button for the Menu List to open or close. */
+      const getButton = (isOpen) => (
+        <MenuButton
+          sx={styles.menuButton}
+          backgroundColor={isOpen ? "ui.link.primary-05 !important" : "unset"}
+        >
+          {showLabel && (
+            <>
+              <span style={{ paddingRight: "8px" }}>{labelText}</span>
+              <Icon
+                name="arrow"
+                iconRotation={isOpen ? "rotate180" : "rotate0"}
+                size="xsmall"
+              />
+            </>
+          )}
+          {!showLabel && (
+            <Icon name="arrow" size={showBorder ? "small" : "medium"} />
+          )}
+        </MenuButton>
+      );
+
+      /** Renders all Menu Items to be passed to the Menu List. Flattened into a reduce() so that
+       Groups of items display together. */
+      const getMenuElements = (data: ListItemsData[] = [], selectedItem) =>
+        data.reduce((lst, item) => {
+          if (item.type === "divider") {
+            // If item is a divider
+            return [
+              ...lst,
+              <MenuDivider sx={styles.dividerItem} key={item.id} />,
+            ];
+          }
+          if (item.type === "group") {
+            // If item is a group: adds group label then passes children
+            return [
+              ...lst,
+              <MenuGroup
+                key={item.id}
+                sx={styles.groupItem}
+                title={item.label}
+              />,
+              ...getMenuElements(item.children, selectedItem),
+            ];
+          }
+          // If item is an action item
+          const isSelected = selected === item.id;
+          const menuItem = (
+            <MenuItem
+              key={item.id}
+              isFocusable={true}
+              data-testid={isSelected ? "selected-item" : ""}
+              onClick={() => handleSelect(item.id, item.onClick)}
+              sx={{
+                ...styles.actionItem,
+                ...(isSelected && styles.selected),
+              }}
+            >
+              <>
+                <div
+                  style={{
+                    display: "flex",
+                    alignItems: "center",
+                    gap: "8px",
+                  }}
+                >
+                  {renderMedia(item.media)}
+                  <span>{item.label}</span>
+                </div>
+              </>
+            </MenuItem>
+          );
+
+          return [...lst, menuItem];
+        }, []);
+
+      return (
+        <ChakraMenu id={id} autoSelect={false} {...rest}>
+          {({ isOpen }) => (
+            <Flex
+              flexDirection={listAlignment === "right" ? "row-reverse" : "row"}
+            >
```

Comment:
@bigfishdesign13 had some doubts about how this alignment works so just flagging this